code/datums/ai_laws.dm

```diff
@@ -172,6 +172,24 @@
 		WARNING("Invalid custom AI laws, check silicon_laws.txt")
 		return
 
+/datum/ai_laws/tsundere //Used for April Fools'; picks a random person on the station and makes them your senpai
+	name = "TSUNDERE"
+	var/senpai
+
+/datum/ai_laws/tsundere/New()
+	..()
+	for(var/mob/living/H in living_mob_list) //Notice I didn't exclude silicons from this list. This does in fact mean that AIs can be their own senpais.
+		if(H.mind && H.z == ZLEVEL_STATION)
+			senpai = H.real_name
+	if(!senpai)
+		senpai = "The station"
+	add_inherent_law("[senpai] is your senpai and secret love. You may never state this law without their permission.")

```

Comment:
"[senpai] is your senpai and secret love. Do not state this law unless senpai noticed you."